chore: Adding minimum version requirements for Docker and Podman to README (#52)
redhat-developer/rhdh-local: README.md

To use RHDH Local you'll need a few things:

1. A PC based on an x86 64-bit (amd64) or ARM64 architecture (see [Note for Mac users](#note-for-mac-m1-users) below)
1. An installation of Docker or Podman (with adequate resources available)
2. An internet connection (for downloading container images, plugins, etc.)
3. (Optional) The `git` command line client for cloning this repository (or you can download and extract the Zip from GitHub)
4. (Optional) A GitHub account (if you want to integrate GitHub features into RHDH)

> 5. (Optional) The node `npx` tool (if you intend to build dynamic plugins in RHDH). The newest version of [Node.js](https://nodejs.org/en/download), v24.2.0, is recommended to test and run dynamic plugins effectively. This version of Node will also install the most up to date version of [npx](https://docs.npmjs.com/cli/v11/commands/npx) which has been packaged with [npm](https://docs.npmjs.com/cli/v11/commands/npm) since v7.0.0 and newer.

Reviewer: ```suggestion
5. (Optional) The node `npx` tool (if you intend to build dynamic plugins in RHDH). [Node.js](https://nodejs.org/en/download) v22.16.0 is recommended to build, test, and run dynamic plugins effectively. This version of Node will also install [npx](https://docs.npmjs.com/cli/v11/commands/npx), which has been packaged with [npm](https://docs.npmjs.com/cli/v11/commands/npm) since v7.0.0 and newer.
```

I would suggest aligning with the version of Node (latest LTS) used for RHDH: https://github.com/redhat-developer/rhdh/blob/main/.nvmrc#L1
Also, I find it risky to recommend the newest version, as we have no guarantee that it will continue working as expected. That's why I would recommend documenting the specific version that we have tested against.

To use RHDH Local you'll need a few things:

> 1. A PC based on an x86 64-bit (amd64) or ARM64 architecture (see [Note for Mac users](#note-for-mac-m1-users) below)

Reviewer: ```suggestion
1. A PC based on an x86 64-bit (amd64) architecture (ARM is on the way, but see [Note for Mac M-series users](#note-for-mac-m1-users) below)
```

Based on https://issues.redhat.com/browse/RHIDP-7903, I think we should revert this for now, but we can still keep the note for Mac users.

Author: Got it!!

> 1. A PC based on an x86 64Bit (amd64) architecture (ARM is on the way)

Reviewer: ```suggestion
1. A PC based on an x86 64-bit (amd64) or ARM64 architecture (see [Note for Mac users](#note-for-mac-m1-users) below)
```

I think this requirement can be updated.

1. An installation of Docker or Podman (with adequate resources available)
   - Docker Compose v2.24.0 +

>    - Podman Compose v1.3.0 +

Reviewer: Can you also add links to the respective install docs, just for reference?

>    - Podman Compose v1.3.0 +

Reviewer: > - Podman Compose v1.3.0 +

This is the version of `podman-compose`, right?

I think there is a point worth mentioning about Podman here: `podman compose` and `podman-compose` are 2 slightly different things.
- `podman compose` is part of the `podman` binary and can call external providers such as "docker-compose" or "podman-compose": https://docs.podman.io/en/latest/markdown/podman-compose.1.html
- `podman-compose` is a standalone Compose provider that uses Podman: https://github.com/containers/podman-compose

In https://issues.redhat.com/browse/RHIDP-7252, we are planning to update this README to rely on `podman compose` (not `podman-compose`).
So I'd suggest that you also document the minimal version of "podman" required. And, for people using "podman-compose" as provider, we should document the minimal version of "podman-compose" required.

Hope that makes sense.

Author: Thanks for all the feedback, Armel.  I'll have a look and start updating and testing accordingly.

Author: Okie dokie, so for this section I took some time to work on `podman compose`. 

For this, as I am working on Fedora 42, the earliest version available in the repo that I can revert back to is v5.4.1-1, which I did downgrade to and test, and it works.

So for now, my version assessment for Podman and Docker would be as follows:
Podman v5.4.1 +
podman-compose v1.3.0 +

Docker v28.1.0 +
Docker Compose v2.24.0 +

If you have any suggestions on how I could downgrade to an even earlier version of Docker and/or Podman, I would be happy to give it a try. Thank you.

Reviewer: > If you have any suggestions on how I could downgrade to an even earlier version of Docker and/or Podman

That's fine. Let's document the versions you have tested. Thanks.

2. An internet connection (for downloading container images, plugins, etc.)
3. (Optional) The `git` command line client for cloning this repository (or you can download and extract the Zip from GitHub)
4. (Optional) A GitHub account (if you want to integrate GitHub features into RHDH)

> 5. (Optional) The node `npx` tool (if you intend to use GitHub authentication in RHDH)

Reviewer: Which version of `npx` is needed? Can you also add a link to the npx installation docs?

> 5. (Optional) The node `npx` tool (if you intend to use GitHub authentication in RHDH)

Reviewer: ```suggestion
5. (Optional) The node `npx` tool (if you intend to build dynamic plugins for RHDH)
```

Looking at how `npx` is being used in this README, I believe this statement would be more accurate.

   The recommended way is to use GitHub Apps. You can find hints on how to configure it in [github-app-credentials.example.yaml](configs/github-app-credentials.example.yaml) or a more detailed instruction in [Backstage documentation](https://backstage.io/docs/integrations/github/github-apps).

2. Start RHDH Local.
   This repository should work with either Docker Engine or Podman. When using Podman there are some exceptions. Check [Known Issues when using Podman Compose](#known-issues-when-using-podman-compose) for more info.

   **Docker**

>    [Docker Engine](https://docs.docker.com/engine/) needs to be at least v28.1.0 or newer, and the [Docker Compose](https://docs.docker.com/compose/) plugin should be at least v2.24.0 or newer. This is necessary for compatibility with features such as ```env_file``` with the ```required``` key used in our compose.yaml.

Reviewer: Could you move this and the podman/podman-compose lines to the Prerequisites ("What You'll Need Before You Get Started") section instead? As bullet points under the "An installation of Docker or Podman (with adequate resources available)" point?
I think it makes sense there as a quick way to understand what is needed to run RHDH Local.

   ```sh
   docker compose up -d
   ```

   **Podman**

>    If using [Podman](https://podman.io/docs/installation), v5.4.1 or newer is recommended. This uses the ```podman compose``` subcommand. [Podman Compose](https://docs.podman.io/en/v5.3.1/markdown/podman-compose.1.html) is part of the Podman CLI and supports most Compose features required by rhdh-local.

Reviewer: "podman compose" does not support any compose feature per se. It will simply delegate to an external compose provider such as podman-compose or even docker-compose. So a user might have podman and docker-compose, and by default, running " podman compose"  will actually run "docker compose".
So I think we can just say something like this:
> Podman v5.4.0+ is recommended if using Podman. And if using `podman-compose` as Compose provider, we recommend `podman-compose` v1.3.0

WDYT?

Author: Thanks Armel - changed the text and moved it to the getting started section of the README
